### ISSUE TITLE
I detect nothing when using my own dataset to train.

### ISSUE THREAD

**Issue by ZoeSnow** [hizhangp/yolo_tensorflow#96]
I trained on my own dataset and replaced the yolo_small.ckpt with data,meta and index files, but no objects could be detected.（好桑心） Does anyone know what I should do？Thank you!!!

**Comment by Yusei37** [comment]
> I trained on my own dataset and replaced the yolo_small.ckpt with data,meta and index files, but no objects could be detected.（好桑心） Does anyone know what I should do？Thank you!!!

I have the same problem, too. Did you solve it? 
I trained it without yolo_small.ckpt, and the code run normally.

**Comment by ZoeSnow** [comment]
I haven't solved the error yet. My training process is also normal but I can't detect anything.

**Comment by zuowk** [comment]
the loss layer in the training file seems to be wrong,so training on your your dataset can't work

**Comment by ZoeSnow** [comment]
@zuowk

Thank you for your reply. Did you train on your own dataset successfully? Could you tell me how to rectify the loss function?

**Comment by JokerKangS** [comment]
> the loss layer in the training file seems to be wrong,so training on your your dataset can't work
I trained it on the pascal_voc dataset without yolo_small.ckpt, and still can't detect anything, too.

**Comment by JokerKangS** [comment]
> the loss layer in the training file seems to be wrong,so training on your your dataset can't work

I trained it on the pascal_voc dataset without yolo_small.ckpt, and still can't detect anything, too.

**Comment by Lizhihzzx** [comment]
I have the same problem.Did anybody solve it?Or it means that i must use yolo_small.ckpt?

**Comment by JokerKangS** [comment]
I found the reason！when i review the origin paper of yolo，i found that the auther pretrain the first 20 convonlutional layers on the ImageNet 1000-class dataset. So i pretrain the first conv-layers on another classification dataset(because i have no imagenet dateset)，then train the whole network. Finally, it can detect things correctly! In short, if you want to train yolo from scratch, you should pretrain the first 20 conv-layers with a classification task
